File: 1742-widest-vertical-area-between-two-points-containing-no-points/widest-vertical-area-between-two-points-containing-no-points.cpp
class Solution {
    bool cmp(vector<int>vec1,vector<int>vec2){
        return vec1[0]-vec2[1];
    }
public:
    int maxWidthOfVerticalArea(vector<vector<int>>& points) {
        int ans=0;
        sort(points.begin(),points.end());
        for(int i=1;i<points.size();i++){
            auto p1=points[i-1];
            auto p2=points[i];
            ans=max(ans,p2[0]-p1[0]);
        }
        return ans;
    }
};

File: 1742-widest-vertical-area-between-two-points-containing-no-points/widest-vertical-area-between-two-points-containing-no-points.java
class Solution {
    public int maxWidthOfVerticalArea(int[][] points) {
        int ans=0;
        Arrays.sort(points,(a,b)->a[0]-b[0]);
        for(int i=1;i<points.length;i++){
            int p1=points[i-1][0];
            int p2=points[i][0];
            ans=Math.max(ans,p2-p1);
        }
        return ans;
    }
}